Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=a1afd01c175324656d0e8f1c82ea94b474953c04
Commit:     a1afd01c175324656d0e8f1c82ea94b474953c04
Parent:     4e14e833ac3b97a4aa8803eea49f899adc5bb5f4
Author:     Joerg Roedel <joerg.roedel@amd.com>
AuthorDate: Tue Nov 18 12:44:21 2008 +0100
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Tue Nov 18 14:52:46 2008 +0100

    x86: default to SWIOTLB=y on x86_64
    
    Impact: fixes korg bugzilla 11980
    
    A kernel for a 64bit x86 system should always contain the swiotlb code
    in case it is booted on a machine without any hardware IOMMU supported
    by the kernel and more than 4GB of RAM. This patch changes Kconfig to
    always compile swiotlb into the kernel for x86_64.
    
    Signed-off-by: Joerg Roedel <joerg.roedel@amd.com>
    Cc: stable@kernel.org
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

 arch/x86/Kconfig |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 93224b5..669c6d5 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -566,7 +566,7 @@ config AMD_IOMMU
 
 # need this always selected by IOMMU for the VIA workaround
 config SWIOTLB
-	bool
+	def_bool y if X86_64
 	help
 	  Support for software bounce buffers used on x86-64 systems
 	  which don't have a hardware IOMMU (e.g. the current generation
